docs/upgrading_to_nameko_3.rst

```diff
@@ -0,0 +1,347 @@
+
+**Optional** — Use the global `config` object to simplify service setup.
+The global is available at all stages of the service lifecycle, including import time, which can make it much simpler to configure services.
+
+**Required ⚠** — Don't pass `config`` to `ServiceRpcProxy` or `ClusterRpcProxy`
+If you are using these standalone RPC clients from the `nameko.standalone` package, note that they no longer accept `config` as a parameter.
+
+You must set up config in the global context before using them, for example with `nameko.config.patch`.
+
+.. _using-config-in-tests:
+
+Using config in tests
+.....................
+
+**Required ⚠** — Read global `config` rather than any of the `x_config`` fixtures.
+There is a breaking change to the way that the `rabbit_config`, `test_config` and `empty_config` fixtures work. They no longer return the config dictionary.
+
+If you are using one of these fixtures, see the recommended pattern in the detailed breakdown of these configuration changes.
+
+**Optional** — Don’t pass config to container_factory or runner_factory.
+If you are using these pytest fixtures, note that the config argument is now deprecated.
+
+See the detailed breakdown of these changes for more information.
+
+.. _using-custom-runners:
+
+Using custom runners
+....................
+
+Required ⚠ — Don’t pass config to ServiceContainer or ServiceRunner
+In Nameko 3, the ServiceContainer and ServiceRunner classes do not accept a config argument anymore. If you are programmatically creating a service runner, rather than using nameko run in the CLI, you must set up config in the global context before using them, for example with nameko.config.patch.
+
+See the detailed breakdown of these changes for more information.
+
+Built-in RPC clients
+....................
+
+Note that the following dependency providers have been renamed:
+
+* `ClusterRpcProxy` -> `ClusterRpcClient`
+* `ServiceRpcProxy` -> `ServiceRpcClient`
+
+The old names are preserved for backwards compatibility.
+
+Upgrading Extensions
+~~~~~~~~~~~~~~~~~~~~
+
+Nameko 3 is backwards-compatible so that extensions written for Nameko 2 should not require any modification in order to work with Nameko 3.
+
+In particular:
+
+The `container_factory` and `runner_factory` fixtures still accept config, even as a deprecated argument. The `ServiceContainer.config` attribute still returns a reference to the config object. This is also deprecated and will raise a warning.
+Before Nameko 2 is sunsetted, extensions should make the necessary adjustments to read from the global config object.
+
+Test Upgrades
+.............
+
+Although extensions will be backwards-compatible when used in services running Nameko 3, some changes may be required to their tests. These are the same changes that apply when :ref:`reading config inside tests for service code <using-config-in-tests>`.
```

Comment:
It is Nameko that is backwards compatible with the extensions, not the reverse. Please reformulate as:

> Although Nameko 3 is backwards-compatible with Nameko 2 extensions, some changes [...]